pages/A.js

```diff
@@ -116,24 +116,25 @@ export class A extends Component {
     this.setState({ width: window.innerWidth });
   }
 
-  setURL = state => {
-    let href = "/A?section=" + state.section;
-    if (Object.keys(state.selectedNeeds).length > 0) {
-      href += "&selectedNeeds=" + Object.keys(state.selectedNeeds).join();
+  setURL = () => {
+    let href = "/A?section=" + this.state.section;
+    if (Object.keys(this.state.selectedNeeds).length > 0) {
+      href += "&selectedNeeds=" + Object.keys(this.state.selectedNeeds).join();
     }
     ["patronType", "serviceType", "statusAndVitals"].forEach(selection => {
-      if (state.selectedEligibility[selection] !== "") {
-        href += `&${selection}=${state.selectedEligibility[selection]}`;
+      if (this.state.selectedEligibility[selection] !== "") {
+        href += `&${selection}=${this.state.selectedEligibility[selection]}`;
       }
     });
+    href += "&lng=" + this.props.t("current-language-code");
     Router.push(href);
   };
 
   setSection = section => {
     let newState = this.state;
     newState.section = section;
     this.setState(newState);
-    this.setURL(newState);
+    this.setURL();
```

Comment:
I don't think this is safe. From what I've read, after calling `this.setState` it is not guaranteed that the state will be set before the next instruction (ie setState works asynchronously). So if the url is set before `setState` finishes, then the url will be set incorrectly and not fixed when `setState` finally finishes running. This is why I coded it here in this rather cumbersome way.

Reply:
Oh okay. I'll change it back